### DISCUSSION TITLE
Fatal error: fault on MacOS 13 Ventura

### DISCUSSION THREAD

**Discussion by hironichu** [projectdiscovery/nuclei#2235] [General]
### Current Behavior: 
Nuclei is crashing on start when running with latest version
the error seems to be "random" can be a Bus error or a SIGSEGV 
updating to latest stable/beta of Go doesn't change much.

EDIT: I must also say, the previous version was working perfectly fine. (2.7.1)

### Expected Behavior:
Nuclei should just show the help section.

### Steps To Reproduce:
1. Get macos 13 ventura beta 1,
2. install latest Nuclei version
3. run and see the crash.

### Anything else:
This was taken before updating to Go 1.19 beta 1
<img width="1667" alt="image" src="https://user-images.githubusercontent.com/64388064/173212409-6c35d302-5373-4bdc-b533-aa85a3fd0fa4.png">

This crash is different and occured after installing Go 1.19 beta 1
<img width="1639" alt="image" src="https://user-images.githubusercontent.com/64388064/173212590-b6aba981-642a-462b-b8c4-01d0b16911af.png">

**Comment by hironichu** [comment]
After updating to Beta 2 and latest dev version of nuclei i get this different error..
```bash

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.7.3

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Your current nuclei-templates v9.0.6 are outdated. Latest is v9.0.8
[INF] Downloading latest release...
panic: close of closed channel

goroutine 1 [running]:
net/http.(*http2clientStream).abortStreamLocked.func1()
	/usr/local/go/src/net/http/h2_bundle.go:7090 +0x51
sync.(*Once).doSlow(0x4cb3626?, 0xc002869930?)
	/usr/local/go/src/sync/once.go:68 +0xc2
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
net/http.(*http2clientStream).abortStreamLocked(0xc0000f6780?, {0x5bd0280?, 0xc000110b10?})
	/usr/local/go/src/net/http/h2_bundle.go:7088 +0x7a
net/http.(*http2clientStream).abortStream(0xc0000f67b0?, {0x5bd0280?, 0xc000110b10?})
	/usr/local/go/src/net/http/h2_bundle.go:7084 +0xac
net/http.http2transportResponseBody.Close({0xc00077f408?})
	/usr/local/go/src/net/http/h2_bundle.go:9181 +0x187
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).downloadReleaseAndUnzip(0xc0002a6140, {0x5bd9368?, 0xc00012a010?}, {0xc0002b6540, 0x5}, {0xc000038d70, 0x4d})
	/Users/ekko/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.6.4-0.20220621191508-7f28c048a601/internal/runner/update.go:318 +0x69d
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).updateTemplatesWithVersion(0xc0002a6140, {0x9, 0x0, 0x8, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}}, ...)
	/Users/ekko/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.6.4-0.20220621191508-7f28c048a601/internal/runner/update.go:169 +0x770
github.com/projectdiscovery/nuclei/v2/internal/runner.(*Runner).updateTemplates(0xc0002a6140)
	/Users/ekko/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.6.4-0.20220621191508-7f28c048a601/internal/runner/update.go:147 +0xf92
github.com/projectdiscovery/nuclei/v2/internal/runner.New(0x65890a0)
	/Users/ekko/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.6.4-0.20220621191508-7f28c048a601/internal/runner/runner.go:92 +0x158
main.main()
	/Users/ekko/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.6.4-0.20220621191508-7f28c048a601/cmd/nuclei/main.go:42 +0x1cc
```


I must also add that while installing Nuclei, i get this message : 
```ld: warning: -no_pie is deprecated when targeting new OS versions
ld: warning: non-standard -pagezero_size is deprecated when targeting macOS 13.0 or later
ld: warning: disabling chained fixups because of unaligned pointers in '_go.func.*'```

**Comment by ehsandeep** [comment]
@hironichu thanks for sharing these details, while we continue to investigate this, just wanted to let you know that nuclei is built based on **Go 1.18** so we do not suggest using `Go 1.19` to build the nuclei, instead either you can download the compiled binary or rollback to `Go 1.18` to build from source.

**Comment by Mzack9999** [comment]
I couldn't reproduce this with the latest beta of go 1.19. I think random crashes can't be excluded with candidate releases. I suggest sticking to go1.18 for the time being until a stable version of go1.19 gets released.

**Reply by hironichu** [reply]
:,) but I wasn't using Go 1.19 it doesn't work in any version of Go on latest macos beta..

I tried because I wanted to know if it was related or not!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
